cardano-crypto-class/src/Cardano/Crypto/SECP256K1/Constants.hsc

```diff
@@ -1,19 +1,46 @@
 {-# LANGUAGE DataKinds #-}
 module Cardano.Crypto.SECP256K1.Constants (
-    SECP256K1_PUBKEY_BYTES,
-    SECP256K1_PRIVKEY_BYTES,
-    SECP256K1_XONLY_PUBKEY_BYTES,
-    SECP256K1_KEYPAIR_BYTES,
+    SECP256K1_ECDSA_PRIVKEY_BYTES,
+    SECP256K1_ECDSA_SIGNATURE_BYTES,
+    SECP256K1_ECDSA_SIGNATURE_BYTES_INTERNAL,
+    SECP256K1_ECDSA_PUBKEY_BYTES,
+    SECP256K1_ECDSA_PUBKEY_BYTES_INTERNAL,
+    SECP256K1_ECDSA_MESSAGE_BYTES,
+    SECP256K1_SCHNORR_PUBKEY_BYTES,
+    SECP256K1_SCHNORR_PRIVKEY_BYTES,
+    SECP256K1_SCHNORR_PUBKEY_BYTES_INTERNAL,
+    SECP256K1_SCHNORR_KEYPAIR_BYTES,
     SECP256K1_SCHNORR_SIGNATURE_BYTES
     )  where
 
 #include <secp256k1.h>
 #include <secp256k1_extrakeys.h>
 
+-- ECDSA-related constants
+
+type SECP256K1_ECDSA_PRIVKEY_BYTES = 32 
+-- As we do not want to serialize the internal state used by ECDSA directly, we
+-- define _two_ values: one for the 'external' representation size, and one for
+-- the 'internal' representation size.
+type SECP256K1_ECDSA_PUBKEY_BYTES = 33
```

Comment:
Maybe a totally stupid question because I don't know a lot about cryptography, but this number `33` looks odd to me. Shouldn't it be `32`? @kozross @lehins @michaelpj

Reply:
disregard, I saw: https://github.com/bitcoin-core/secp256k1/blob/master/include/secp256k1.h

Reply:
It indeed is odd ;) no, but out of jokes, it is correct. 33 is expected, because a public key is an elliptic curve point which contains two coordinates `(x, y)`. Each coordinate takes 32 bytes. However, given an `x` coordinate, there is only two possible values for `y`. Therefore, the way a public key is serialised is 1 byte to represent which of the two `y` values we use, and 32 bytes for the `x` coordinate, therefore, 33 bytes 👍  The same does not happen for Schnorr, which only uses the `x` coordinate and _always_ assumes that the `y` coordinate is even. Therefore, Schnorr public keys (as described in BIP 340) take only 32 bytes.